pkg/apis/build/v1alpha2/image_validation.go

```diff
@@ -45,8 +45,12 @@ func (i *Image) SetDefaults(ctx context.Context) {
 		i.Spec.SuccessBuildHistoryLimit = &defaultSuccessfulBuildHistoryLimit
 	}
 
-	if i.Spec.CacheSize == nil && ctx.Value(HasDefaultStorageClass) != nil {
-		i.Spec.CacheSize = &defaultCacheSize
+	if i.Spec.Cache == nil && ctx.Value(HasDefaultStorageClass) != nil {
```

Comment:
Is this correct? Should we check for `i.Spec.Cache.Request != nil` as well? What if it uses `registryCache`?

Reply:
I think this is right!